### ISSUE TITLE
Contract shouldn't try to match meta-attributes in mappings and tests for xml

### ISSUE THREAD

**Issue by vkochnev** [spring-cloud/spring-cloud-contract#1571]
**Describe the bug**
On version 3.0.0-M5 (M4 and M3 also looks affected) it is not possible to write contracts for xml requests/responses using namespaces. The reason is there are matchers automatically added trying to match `xmlns` and `xmlns:prefix` attributes, which is impossible, because they are not actual attributes and cannot be queried with xPath.

**Sample**
Having contract
```
description: Test
request:
  method: POST
  url: /test
  headers:
    Content-Type: application/xml
  body: |
    <request xmlns="urn:example.com/Test">
        <childOne>one</childOne>
        <childTwo>two</childTwo>
    </request>
response:
  status: 200
  headers:
    Content-Type: text/xml; charset=utf-8
  body: |
    <tst:response xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:tst="urn:example.com/Test">
        <result xsi:type="tst:result" xmlns="urn:example.com/Test">
            <childOne xsi:type="xsd:string">one</childOne>
            <childTwo xsi:type="xsd:string">two</childTwo>
        </result>
    </tst:response>
```
We get several erroneous matchers generated
in mappings
```
"matchesXPath" : {
    "expression" : "/request/@xmlns",
    "equalTo" : "urn:example.com/Test"
}
```
and in test
```
assertThat(valueFromXPath(parsedXml, "/tst:response/@xmlns:tst")).isEqualTo("");
assertThat(valueFromXPath(parsedXml, "/tst:response/@xmlns:xsd")).isEqualTo("");
assertThat(valueFromXPath(parsedXml, "/tst:response/@xmlns:xsi")).isEqualTo("");
assertThat(valueFromXPath(parsedXml, "/tst:response/result/@xmlns")).isEqualTo("");
```

**Comment by marcingrzejszczak** [comment]
https://github.com/spring-cloud/spring-cloud-contract/issues/1517 is this the same / related?

**Comment by vkochnev** [comment]
Not exactly the same, but might be related. #1517 is about generating xPath queries in general when specifying namespaces, this one is about matching `xmlns` attributes themselves. I mean they will arise at the same time, but I suspect they might be fixed separately. And unlike #1517, I don't know of a workaround for this one, cause matchers cannot be overridden or suppressed manually.

**Comment by bono007** [comment]
I should have a pull request submitted w/ a fix for this and #1517 w/in next 24 hours. I have the explicit namespace matching fixed and need to work on the default namespace case.
